extensions/lua/LuaScriptStateManager.cpp

```diff
@@ -0,0 +1,54 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "LuaScriptStateManager.h"
+
+namespace org::apache::nifi::minifi::extensions::lua {
+
+namespace {
+core::StateManager::State from_lua(const sol::table& lua_state) {
+  core::StateManager::State cpp_state;
+  for (const auto& [lua_state_key, lua_state_value] : lua_state)
+    cpp_state[lua_state_key.as<std::string>()] = lua_state_value.as<std::string>();
```

Comment:
What if we get back something other than a string? Is it handled gracefully?

Reply:
That would cause the onTrigger to fail with an error message from the interpreter, something like this:
```
  stack index -1, expected string, received number
stack traceback:
	[C]: in method 'set'
	[string "..."]:12: in function 'base.onTrigger'
```